PR: https://github.com/odoo/odoo/pull/218157

From: 6ca20e73120905626eec468038cdb07088f2d256
From: rame-odoo
Date: 2026-02-24 21:03:07

Breaking data model changes scores: del:2 + add:0, change matches:
-    _inherit = 'mail.activity.plan.template'
-    responsible_type = fields.Selection(

Total Changes: 38

[IMP] hr, *: employee launch plan should work even if not fully configured

* = mail, hr_fleet, hr_sign

Employee launch plans should work even if employee not well configured.
Should not block if manager/fleet/user not set, just warn and assign to current
user.

This update also removes responsible types 'coach' and 'fleet_manager'
and cleans up all related logic, validations, and references.

This PR includes:
* Changed blocking errors to warnings when manager/user not set.
* Updated warning message strings to be shorter and clearer.
* Removed 'coach' and 'fleet_manager' responsible types and associated codes.
* Adjut some tet cases accordingly.

Related Enterprise PR: https://github.com/odoo/enterprise/pull/99355
Related Upgrade PR: https://github.com/odoo/upgrade/pull/9309

Task - 4879248

closes odoo/odoo#218157

Related: odoo/enterprise#99355
Related: odoo/upgrade#9309
Signed-off-by: Yannick Tivisse (yti) <yti@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_fleet/models/__init__.py
+++ b/addons/hr_fleet/models/__init__.py
@@ -9,4 +9,3 @@ from . import fleet_vehicle_log_contract
 from . import fleet_vehicle_log_services
 from . import fleet_vehicle_odometer
 from . import ir_attachment
-from . import mail_activity_plan_template

--- a/addons/hr_fleet/models/mail_activity_plan_template.py
+++ b/None
@@ -1,37 +0,0 @@
-# -*- coding: utf-8 -*-
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import api, fields, models, _
-from odoo import exceptions
-
-
-class MailActivityPlanTemplate(models.Model):
-    _inherit = 'mail.activity.plan.template'
-
-    responsible_type = fields.Selection(
-        selection_add=[('fleet_manager', "Fleet Manager")],
-        ondelete={'fleet_manager': 'set default'})
-
-    @api.constrains('plan_id', 'responsible_type')
-    def _check_responsible_hr_fleet(self):
-        """ Ensure that hr types are used only on employee model """
-        for template in self.filtered(lambda tpl: tpl.plan_id.res_model != 'hr.employee'):
-            if template.responsible_type == 'fleet_manager':
-                raise exceptions.ValidationError(_("Fleet Manager is limited to Employee plans."))
-
-    def _determine_responsible(self, on_demand_responsible, on_demand_responsible_fname, employee):
-        if self.responsible_type == 'fleet_manager' and self.plan_id.res_model == 'hr.employee':
-            employee_id = self.env['hr.employee'].browse(employee._origin.id)
-            vehicle = employee_id.car_ids[:1]
-            error = False
-            warning = False
-            if not vehicle:
-                error = _('Employee %s is not linked to a vehicle.', employee_id.name)
-            if vehicle and not vehicle.manager_id:
-                warning = _("The vehicle of employee %(employee)s is not linked to a fleet manager, assigning to you.", employee=employee_id.name)
-            return {
-                'responsible': vehicle.manager_id or self.env.user,
-                'error': error,
-                'warning': warning
-            }
-        return super()._determine_responsible(on_demand_responsible, on_demand_responsible_fname, employee)
